src/docker/mod.rs

```diff
@@ -0,0 +1,40 @@
+mod engine;
+mod local;
+mod shared;
+
+use self::local::*;
+
+pub use self::engine::*;
+pub use self::shared::*;
+
+use std::path::Path;
+use std::process::ExitStatus;
+
+use crate::cargo::CargoMetadata;
+use crate::errors::*;
+use crate::{Config, Target};
+
+#[allow(clippy::too_many_arguments)] // TODO: refactor
+pub fn container_run(
```

Comment:
Rename `run` to `container_run`, so we prefix it correctly since we'll need to use glob imports or very lengthy imports in the future.